src/rgw/motr/gc/gc.h

```diff
@@ -0,0 +1,66 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=2 sw=2 expandtab ft=cpp
+
+/*
+ * Garbage Collector Classes for the CORTX Motr backend
+ *
+ * Copyright (C) 2022 Seagate Technology LLC and/or its Affiliates
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation. See file COPYING.
+ *
+ */
+
+#ifndef __MOTR_GC_H__
+#define __MOTR_GC_H__
+
+#include "rgw_sal_motr.h"
+#include "common/Thread.h"
+#include <mutex>
+#include <condition_variable>
+#include <atomic>
+
+class MotrGC : public DoutPrefixProvider {
+ private:
+  CephContext *cct;
+  rgw::sal::Store *store;
```

Comment:
This should be MotrStore, have you figured out the error?

Reply:
not yet.
Even though MotrGC might never use another store than motr, keeping Store* is good practice I think, and everything should work with it. We can spend time on fixing it if it becomes an obstacle.

Reply:
Need to typecast explicitly.

Reply:
Yes, so the Store implementation will have to down-cast the pointer explicitly to call its own methods.
Is there something wrong with it?

Reply:
Typecasting won't be expensive here, but if we can avoid it, we should avoid it.

Reply:
sure. yes, let's find some way. I agree that it is not the best way.